src/sentry/web/frontend/auth_organization_login.py

```diff
@@ -136,6 +140,13 @@ def handle(self, request, organization_slug):
         except AuthProvider.DoesNotExist:
             auth_provider = None
 
+        if request.COOKIES.get('session_expired'):
+            messages.add_message(request, messages.WARNING, ERR_EXPIRED)
+
         if not auth_provider:
-            return self.handle_basic_auth(request, organization)
-        return self.handle_sso(request, organization, auth_provider)
+            response = self.handle_basic_auth(request, organization)
+        else:
+            response = self.handle_sso(request, organization, auth_provider)
+
+        response.delete_cookie('session_expired')
```

Comment:
It will look like this:

```python
has_session_expired = 'session_expired' in request.COOKIES
if has_seession_expired:
  messages.add_message(request, messages.WARNING, WARN_SESSION_EXPIRED)

if has_session_expired:
  response.delete_cookie('session_expired')
```

Okay?